### PR TITLE
[line] fix verbose names in time shift

### PR DESCRIPTION
**Pull request by mistercrunch**
screen shot 2018-04-04 at 10 32 20 pm
closes https://github.com/apache/incubator-superset/issues/4750

### CODE REVIEW

File: superset/assets/spec/javascripts/visualizations/nvd3_viz_spec.jsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import { formatLabel } from '../../../visualizations/nvd3_vis';
+
+describe('nvd3 viz', () => {
+  const verboseMap = {
+    foo: 'Foo',
+    bar: 'Bar',
+  };
+  describe('formatLabel', () => {
+    it('is formats simple labels', () => {
```
Comment (betodealmeida) [inline superset/assets/spec/javascripts/visualizations/nvd3_viz_spec.jsx]: Nit, drop the "is"?

`it('formats simple labels', () => {`

File: superset/assets/visualizations/nvd3_vis.js [hunk 2]
```diff
@@ -83,23 +83,24 @@ function getMaxLabelSize(container, axisClass) {
...
   let label;
-  if (Array.isArray(column) && column.length) {
-    label = verbose_map[column[0]] || column[0];
-    if (column.length > 1) {
-      label += ', ';
+  if (Array.isArray(input) && input.length) {
+    const verboseLabels = input.filter(s => s !== '---').map(verboseLkp);
+    label = verboseLabels.join(', ');
+    if (input.length > verboseLabels.length) {
+      label += '---';
```
Comment (betodealmeida) [inline superset/assets/visualizations/nvd3_vis.js]: Did you test leaving a space between the label and the dashes? Seems a bit cluttered to me without the space there.

**Review by betodealmeida — APPROVED** [review]
Cool cool cool, looks awesome!

**Comment by michellethomas** [comment]
@mistercrunch it looks like this broke the time shift color assignment (that you fixed here https://github.com/apache/incubator-superset/pull/4202), but it seemed like that broke the verbose names in time shift. Any thoughts on how to solve for both cases?

**Pull request merged**